The bot fails to read the gcc compiling error and goes straight to running the output binary.

It returns the following output:

```
Language:
gcc

Code:
This should throw a compiler error

Output:
/piston/packages/gcc/10.2.0/run: line 6: ./a.out: No such file or directory
```

When it should have returned something like:

```
Language:
gcc

Code:
This should throw a compiler error

Output:
main.cpp:1:1: error: ‘This’ does not name a type
 This should throw a compiler error
```

I'm guessing it does this with any other compilable languages?

--- a/cmd/bot/main.go
+++ b/cmd/bot/main.go
@@ -192,13 +192,14 @@ func main() {
 }
 
 var (
-	BlockLanguage = "Language"
-	BlockCode     = "Code"
-	BlockStdin    = "Stdin"
-	BlockOutput   = "Output"
-	BlockError    = "Error"
+	BlockLanguage       = "Language"
+	BlockCode           = "Code"
+	BlockStdin          = "Stdin"
+	BlockCompilerOutput = "Compiler Output"
+	BlockOutput         = "Output"
+	BlockError          = "Error"
 )
-var blockNames = []string{BlockLanguage, BlockCode, BlockStdin, BlockOutput, BlockError}
+var blockNames = []string{BlockLanguage, BlockCode, BlockStdin, BlockCompilerOutput, BlockOutput, BlockError}
 
 func buildOutput(blocks map[string]string) string {
 	var formattedBlocks []string
@@ -231,10 +232,11 @@ func formatPistonResponse(request piston.RunRequest, response piston.RunResponse
 
 	case piston.ResultSuccess:
 		return buildOutput(map[string]string{
-			BlockLanguage: request.Language,
-			BlockCode:     request.Code,
-			BlockStdin:    request.Stdin,
-			BlockOutput:   response.Output,
+			BlockLanguage:       request.Language,
+			BlockCode:           request.Code,
+			BlockStdin:          request.Stdin,
+			BlockCompilerOutput: response.CompilerOutput,
+			BlockOutput:         response.Output,
 		})
 	}
 

--- a/piston.go
+++ b/piston.go
@@ -99,8 +99,9 @@ func CreateRequest(text string) (RunRequest, error) {
 }
 
 type RunResponse struct {
-	Result string
-	Output string
+	Result         string
+	Output         string
+	CompilerOutput string
 }
 
 var (
@@ -172,11 +173,15 @@ func RunCode(request RunRequest) RunResponse {
 		}
 	}
 
-	var data struct{ Run struct{ Output string } }
+	var data struct {
+		Run     struct{ Output string }
+		Compile struct{ Output string }
+	}
 	json.Unmarshal(body, &data)
 
 	return RunResponse{
-		Result: ResultSuccess,
-		Output: data.Run.Output,
+		Result:         ResultSuccess,
+		Output:         data.Run.Output,
+		CompilerOutput: data.Compile.Output,
 	}
 }

